core/src/data/value/literal.rs

```diff
@@ -166,12 +182,12 @@ impl Value {
                 .parse::<i64>()
                 .map(Value::I64)
                 .map_err(|_| ValueError::LiteralCastFromTextToIntegerFailed(v.to_string()).into()),
-            (DataType::Int, Literal::Number(v)) => v
-                .to_f64()
-                .map(|v| Value::I64(v.trunc() as i64))
-                .ok_or_else(|| {
-                    ValueError::UnreachableLiteralCastFromNumberToInteger(v.to_string()).into()
-                }),
+            (DataType::Int, Literal::Number(v)) => match v.to_i64() {
+                Some(x) => Ok(Value::I64(x)),
+                None => {
+                    Err(ValueError::UnreachableLiteralCastFromNumberToInteger(v.to_string()).into())
+                }
+            },
```

Comment:
This change also does not look related to this issue.
We can take this in the different issue or PR.

Reply:
Yes, I found some error while doing some testing and added this.  I'll look into creating an issue/PR about this.